java-bootcamp-2018-tdd/src/main/java/com/globant/business/shoppingCart/ShoppingCart.java

```diff
@@ -0,0 +1,99 @@
+package com.globant.business.shoppingCart;
+
+import java.util.Collection;
+import java.util.Hashtable;
+
+public class ShoppingCart implements IShoppingCart {
+	/**
+	 * Hash whit the rows
+	 */
+	Hashtable<String, RowCart> products;
+
+	/**
+	 * constructor
+	 */
+	public ShoppingCart() {
+		products = new Hashtable<String, RowCart>();
+	}
+
+	/**
+	 * add a new product to the cart whit its quantity if the product is already in
+	 * the cart add the amount
+	 */
+	public Product addProduct(String name, String category, float price, int quantity) {
+		if (!products.containsKey(name)) {
+			RowCart row = new RowCart(new Product(name, category, price), quantity);
+			products.put(name, row);
+		} else {
+			products.get(name).addQuantity(quantity);
+		}
+		return products.get(name).getProduct();
+	}
+
+	/**
+	 * set the amount of one product in the cart if the new quantity is less than 1,
+	 * remove the product
+	 */
+	public Product changeamount(String product, int quantity) {
+		if (quantity > 0) {
+			RowCart current = products.get(product);
+			current.changeAmmount(quantity);
+			return current.getProduct();
+		} else {
+			RowCart current = products.get(product);
+			removeProduct(product);
+			return current.getProduct();
+		}
+	}
+
+	/**
+	 * remove a product of the cart
+	 */
+	public Product removeProduct(String product) {
+		if (products.containsKey(product)) {
+			RowCart current = products.get(product);

```

Comment:
I think this should have an amount to remove.

Reply:
In several online stores, in the cart there is a delete button and a change  amount button. That's why I implemented it with the "changeAmount" and "removeProduct" methods. In change amount, if the new quantity is 0 or lower, the product is removed.